kpm/lib/kpm/nexus_helper/actions.rb

```diff
@@ -0,0 +1,30 @@
+require_relative 'nexus_api_calls_v2'
+#require_relative 'nexus_api_calls_v3'
+
+module KPM
+  module NexusFacade
+    class Actions
+      attr_reader :nexus_api_call
+
+      def initialize(overrides,ssl_verify)
+        #this is where the version is verified
+        #example if
+        #@nexus_api_call = overrides['version'] == '3' ? NexusApiCallsV3.new(overrides, ssl_verify) : NexusApiCallsV2.new(overrides, ssl_verify)
+        @nexus_api_call = NexusApiCallsV2.new(overrides, ssl_verify)
+      end
+
+      def pull_artifact(coordinates, destination=nil)
+        nexus_api_call.pull_artifact(coordinates, destination)
+      end
+
+      def get_artifact_info(coordinates)
+        nexus_api_call.get_artifact_info(coordinates)
```

Comment:
Could you add `remote` tests for that specific file? That way, it would be easy to test just the client piece, once we move to `v3`.

Try to find small artifacts (especially for `pull_artifact`), so it doesn't slow the suite too much. Also, testing various types of `coordinates` would be nice (with/without classifier, etc.)